boards: arm: add board for CPUNET domain

hakonfam opened the pull request:
This configuration is used to describe what board should
be associated with the CPUNET domain.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
boards: arm: add board for CPUNET domain

This configuration is used to describe what board should
be associated with the CPUNET domain.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

tejlmand commented on boards/arm/nrf5340pdk_nrf5340/Kconfig:
```diff
@@ -43,6 +43,12 @@ config BOARD_ENABLE_CPUNET
+	string
+	default "nrf5340pdk_nrf5340_cpunet"
+	help
+	  The name of the BOARD corresponding to the 'NET_CORE' domain.
```
Not sure this is clear to readers (at least it's a little cryptic to me).

How about something like (in case I understand the purpose correctly)
```suggestion
        The board which will be used for NET_CORE domain when creating a multi image application where one or more images should be located on another board. For example hci_rpmsg on the nRF5340_cpunet for Bluetooth applications.
```

tejlmand reviewed (approved):
LGTM.

Please rebase.

ioannisg reviewed (changes requested):
-  Is this Kconfig to be used in the upstream project anywhere?

-  Don't use "core", we either use CPU or MCU for Network/Application 
Use BOARD_ as prefix, if possible 

- Should the option depend on BOARD_ENABLE_CPUNET?

- Fix indentation in help text

- This option needs to be present in the DK definition as well

carlescufi commented:
@hakonfam can you rebase
@ioannisg can you take another look?

hakonfam commented:
> * Is this Kconfig to be used in the upstream project anywhere?

Not for now, I assume it will be useful at a later time.

> * Don't use "core", we either use CPU or MCU for Network/Application

Changed to `DOMAIN_CPUNET_BOARD`
 
>   Use BOARD_ as prefix, if possible

That would indicate that this is a board, which it is not.
IMO the current name works well.

> * Should the option depend on BOARD_ENABLE_CPUNET?

Added this.

> * Fix indentation in help text

Fixed.

> * This option needs to be present in the DK definition as well

What DK definition?
